concepts/numbers/about.md

```diff
@@ -0,0 +1,176 @@
+# About
+
+Lua uses two representations for numbers: 64-bit integer numbers and double-precision floating-point numbers.
+
+~~~exercism/note
+Before version 5.3, Lua represented all numbers using double-precision floating-point format.
+~~~
+
+Numbers can be written with an optional decimal part, plus an optional decimal exponent.
+```lua
+1               --> 1
+0.2             --> 0.2
+0.3e4           --> 2000.0
+4.0e-3          --> 0.004
+5.0e+20         --> 5e+20
+```
+
+A number with a decimal point or an exponent is considered a float and without is treated as an interger, but both have the type 'number'.
+```lua
+type(1)         --> number
+type(2.0)       --> number
+```
+
+If we want to distinguish between floats and intergers, the `type` function of the `math` module  can be used.
+```lua
+math.type(1)    --> integer
+math.type(2.0)  --> float
+```
+
+## Comparsion
+
+Integers and floats with the same value compare as equal:
+```lua
+1 == 1.0        --> true
+-2 == -2.0      --> true
+0.3e3 == 300    --> true
+```
+
+## Arithmetic
+
+Lua supports the usual set of arithmetic operators: additon, subtraction, muliplication, divison and negation. Additional it supports floor division, modulo and exponentiation.
+
+### Addition, substraction, multiplication and negation
+
+If both operands are integer the result of the operations is an integer. The same applies to floats.
+```lua
+1 + 2     --> 3
+1.0 + 2.0 --> 3.0
+```
+
+If one operand is a float, the remaining integers will be converted to floats as well and the operations results in a float.
+```lua
+1.0 + 2   --> 3.0
+```
+
+### Division
+
+Division always operates on floats and gives floats as a result.
+```lua
+1.0 / 2.0 --> 0.5
+2 / 1     --> 2.0
+```
+
+Dividing by zero will result in `inf` or `-inf`. Dividing zero by zero will result in `-nan`.
+```lua
+1 / 0     --> inf
+-1 / 0    --> -inf
+0 / 0     --> -nan
+```
+
+### Floor Divison
+
+If an integer result is needed, floor division can be used, which always rounds the quotient towards minus infinity, ensuring an integral result for all operands.
+
+~~~exercism/note
+Before version 5.3, Lua had no arithmetic operator for floor division.
+~~~
+
+```lua
+3 // 2    --> 1
+6 // 2    --> 3
+```
+
+Floor divison follows the same rules of the addition, substraction, multiplication and negation operators - if both operators are integers the result of the operation is an integer, otherwise the result is a float.
+
+```lua
+3.0 // 2   --> 1.0
+6.0 // 2.0 --> 3.0
+```
+
+When dividing by zero using floor division, Lua will raise an error.
+```lua
+3 // 0     --> stdin:1: attempt to divide by zero
+```
+
+### Modulus
+
+Modulus is used to get the remainder of the division of two operands.
+```lua
+8 % 2      --> 0
+13 % 9     --> 4
+0.3 % 0.2  --> 0.1
+```
```

Comment:
Should we cover negative numbers here? Modulus and remainder differ in how they handle negative numbers, and I believe Lua actually uses remainder, not modulus.

Reply:
The manual refers to it as "modulo" and defines it as the remainder:
https://www.lua.org/manual/5.3/manual.html#3.4.1

Reply:
> Should we cover negative numbers here? Modulus and remainder differ in how they handle negative numbers, and I believe Lua actually uses remainder, not modulus.

added negative number examples - was that what you meant?

Reply:
I think the term "modulus" has a different meaning than remainder. Modulus gives different results for negative numbers.

The Lua manual defines the operator as "modulo" instead of "modulus" and specifies that it gives the remainder. Maybe we should use the same name and explanation.

Reply:
Ah ok. I already changed this, that was a typo :)